lib/java/ydoc-polyfill/src/main/resources/org/enso/ydoc/polyfill/web/zlib.js

```diff
@@ -3,21 +3,36 @@
     class Buffer {
 
         #buffer;
+        #offset;
+        #length;
 
-        constructor(buffer) {
+        constructor(buffer, offset, length) {
```

Comment:
This one is here to support [Buffer.from(arrayBuffer[, byteOffset[, length]])](https://nodejs.org/api/buffer.html#static-method-bufferfromarraybuffer-byteoffset-length). Originally I tried to just extract `offset` and `length` from `buffer`, create a copy of such buffer and pass it into the old constructor. But that doesn't seem to work.

The [Buffer.from(arrayBuffer[, byteOffset[, length]])](https://nodejs.org/api/buffer.html#static-method-bufferfromarraybuffer-byteoffset-length) factory method is supposed to provide **live view** of the buffer. E.g. further changes are still propagated to the buffer.

As such we are extracting the `slice` _lazily_ in `buffer()` getter.